test/TestLogger.UnitTests/TestCaseNameParserTests.cs

```diff
@@ -45,6 +45,36 @@ public class TestCaseNameParserTests
         [DataRow("z.y.x.ape.bar('.',False)", "z.y.x", "ape", "bar('.',False)")]
         [DataRow("z.y.x.ape.bar('\\'',False)", "z.y.x", "ape", "bar('\\'',False)")]
         [DataRow("z.y.x.ape.bar('\\\\',False)", "z.y.x", "ape", "bar('\\\\',False)")]
+
+        // Strip out any line breaks
+        [DataRow("z.y.x.ape.bar('aa\r\nbb',False)", "z.y.x", "ape", "bar('aabb',False)")]
+        [DataRow("z.y.x.ape.bar('aa\nbb',False)", "z.y.x", "ape", "bar('aabb',False)")]
+        [DataRow("z.y.x.ape.bar('aa\rbb',False)", "z.y.x", "ape", "bar('aabb',False)")]
+
+        [DataRow("NetCore.Tests.NetCoreOnly.Issue28_Examples.ExampleTest2(True,4.8m,4.5m,(4.8, False))", "NetCore.Tests.NetCoreOnly", "Issue28_Examples", "ExampleTest2(True,4.8m,4.5m,(4.8, False))")]
+        [DataRow("NetCore.Tests.NetCoreOnly.Issue28_Examples(asdf \".\\).ExampleTest2(True,4.8m,4.5m,(4.8, False))", "NetCore.Tests.NetCoreOnly", "Issue28_Examples(asdf \".\\)", "ExampleTest2(True,4.8m,4.5m,(4.8, False))")]
+        [DataRow("NetCore.Tests.NetCoreOnly.Issue28_Examples(asdf \".\\).ExampleTest2", "NetCore.Tests.NetCoreOnly", "Issue28_Examples(asdf \".\\)", "ExampleTest2")]
+        [DataRow("z.y.X(x", "z", "y", "X(x")]
+        [DataRow("z.y.x(", "z", "y", "x(")]
+        [DataRow("z.y.X\"x", "z", "y", "X\"x")]
+        [DataRow("z.y.x\"", "z", "y", "x\"")]
+        [DataRow("z.y.X\\x", "z", "y", "X\\x")]
+        [DataRow("z.y.x\\", "z", "y", "x\\")]
+        [DataRow("z.y.X\\)", "z", "y", "X\\)")]
+        [DataRow("z.y.X\')", "z", "y", "X\')")]
+        [DataRow("z.y.\'x", "z", "y", "\'x")]
+        [DataRow("z.y.x))", "z", "y", "x))")]
+        [DataRow("z.y.x()x", "z", "y", "x()x")]
+        [DataRow("z.a.b((0,1)", "z", "a", "b((0,1)")]
+        [DataRow("z.a.b((0,1)))", "z", "a", "b((0,1)))")]
+        [DataRow("z.a.b((0,(0,1))", "z", "a", "b((0,(0,1))")]
+        [DataRow("z.a.b((0,(0,1))))", "z", "a", "b((0,(0,1))))")]
+        [DataRow("a.z.y.x.", "a.z", "y", "x.")]
+
+        // These produce strange results but don't output errors. TODO decide if these are ok.
```

Comment:
These are clearly parsing incorrectly, but the string we are parsing is so badly formed i'm not sure its important.

Reply:
May be add a test (in case we don't have one already) to ensure these have a good fallback since the parsing fails.

Reply:
If they truly fail (produce no output) that condition is tested below `Parse_FailsGracefullyOnNonParseableInputs_WithConsoleOutput`

I'm not sure what more we can do to detect that while we did produce a parsed output, that it looks nonsensical.

Reply:
This works!